metaflow/plugins/resources_decorator.py

```diff
@@ -0,0 +1,62 @@
+from metaflow.decorators import StepDecorator
+
+
+class ResourcesDecorator(StepDecorator):
+    """
+    Step decorator to specify the resources needed when executing this step.
+    This decorator passes this information along when requesting resources
+    to execute this step.
+
+    This decorator can be used for two purpose:
+    When using with AWS Batch decorator, only 'cpu', 'gpu', and 'memory' parameters are supported.
+    When using for Kubeflow Pipeline, 'cpu', and 'memory' sets resource requests;
+        'cpu_limit', 'gpu', 'memory_limit' sets resource limit.
+        For more details please refer to
+        https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+    This decorator is ignored if the execution of the step does not happen on Batch.
+    To use, annotate your step as follows:
+    ```
+    @resources(cpu=32, memory="15G")
+    @step
+    def myStep(self):
+        ...
+    ```
+
+    Parameters
+    ----------
+    cpu : Union[int, float, str]
```

Comment:
Why does this `Union` have 3 parameters, when only 2 are described (`AWS Batch` and `KFP`) in the associated documentation?

Reply:
These are the allowed types. Sometimes you can request a friction of CPUs so float is added. For different units you'd better pass in str.